core/storage/learner_group/gae_models.py

```diff
@@ -0,0 +1,297 @@
+                random.choice(string.ascii_lowercase + string.ascii_uppercase)
+                for _ in range(base_models.ID_LENGTH))
+            if not cls.get_by_id(group_id):
+                return group_id
+
+        raise Exception('New id generator is producing too many collisions.')
+
+    @classmethod
+    def create(
+        cls,
+        group_id: str,
+        title: str,
+        description: str
+    ) -> LearnerGroupModel:
+        """Creates a new LearnerGroupModel instance and returns it.
+
+        Args:
+            group_id: str. The ID of the learner group.
+            title: str. The title of the learner group.
+            description: str. The description of the learner group.
+
+        Returns:
+            LearnerGroupModel. The newly created LearnerGroupModel instance.
+
+        Raises:
+            Exception. A learner group with the given group ID exists already.
+        """
+        if cls.get_by_id(group_id):
+            raise Exception(
+                'A learner group with the given group ID exists already.')
+
+        entity = cls(id=group_id, title=title, description=description)
+
+        entity.update_timestamps()
+        entity.put()
+        return entity
+
+    @staticmethod
+    def get_field_names_for_takeout() -> Dict[str, str]:
+        """We do not want to export all user ids in the facilitator_user_ids,
+        student_user_ids and invited_user_ids fields, so we export them as
+        fields only containing the current user's id.
+        """
+        return {
+            'facilitator_user_ids': 'facilitator_user_id',
+            'student_user_ids': 'student_user_id',
+            'invited_user_ids': 'invited_user_id'
+        }
+
+    @classmethod
+    def export_data(cls, user_id: str) -> Dict[str, Dict[str, List[str]]]:
+        """Takeout: Export LearnerGroupModel user-based properties.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. A dict containing the user-relevant properties of
+            LearnerGroupModel.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+        ))
+        user_data = {}
+        for learner_group_model in found_models:
+            # If the user is a student, we export all fields except
+            # facilitator_user_ids, invited_user_ids and the student_user_id
+            # field is exported only containing the current user's id.
+            if user_id in learner_group_model.student_user_ids:
+                user_data[learner_group_model.id] = {
+                    'title': learner_group_model.title,
+                    'description': learner_group_model.description,
+                    'facilitator_user_id': '',
+                    'student_user_id': user_id,
```

Comment:
In general we do not export user IDs, as they should be kept secret. They can be either replaced by username or not exported at all.

Reply:
hmm, I see that exporting user id has not much meaning here, so have removed their export. Thanks!